src/NotesApplication/Services/DBService.cs

```diff
@@ -2,21 +2,98 @@
 using Microsoft.Azure.Cosmos;
 using System.Net.Http;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using ACMTTU.NoteSharing.Shared.SDK.Controllers;
+using ACMTTU.NoteSharing.Platform.NotesApplication.Services;
+using NotesApplication;
 
 namespace ACMTTU.NoteSharing.Platform.NotesApplication.Services
 {
-    public class DBService : PlatformBaseService
+
+    /// <summary>
+    /// An abstract class for database services
+    /// </summary>
+    public interface IDBService
     {
-        public Container container;
-        public DBService(IHttpClientFactory clientFactory) : base(clientFactory)
-        {
-        }
+
+        Task<T> CreateItem<T>(T undatabasedObject);
+        Task<T> ReadItem<T>(string id);
+        Task<ItemResponse<T>> ReplaceItem<T>(T update, string id);
+        Task<ItemResponse<T>> DeleteItem<T>(string id);
+
+    }
+
+    /// <summary>
+    /// An implementation of IDBService to interface with Azure Cosmos DB
+    /// </summary>
+    public class CosmosDBService : PlatformBaseService, IDBService
+    {
+        private Container _container;
+        private PartitionKey _partKey = new PartitionKey("notes");
+
+        public CosmosDBService(IHttpClientFactory clientFactory) : base(clientFactory) { }
```

Comment:
For this, can you revert the name back to DBService? The reasoning behind the name is "What if we want to change database providers?" We might not always use Cosmos. That's why we're abstracting it out into its own class.